pytest_inmanta/plugin.py

```diff
@@ -581,15 +581,49 @@ def apply_filter(resource: Resource) -> bool:
         yield resource
 
 
-class DeployResult:
+def get_resource(
+    resources, resource_type: str, **filter_args: object
+) -> typing.Optional[Resource]:
+    """
+    Get a resource of the given type and given filter on the resource attributes. If multiple resource match, the
+    first one is returned. If none match, None is returned.
+
+    :param resource_type: The exact type used in the model (no super types)
+    """
+
+    def check_serialization(resource: Resource) -> Resource:
+        """Check if the resource is serializable"""
+        serialized = json.loads(json_encode(resource.serialize()))
+        return Resource.deserialize(serialized)
+
+    try:
+        resource = next(get_resources_matching(resources, resource_type, **filter_args))
+        resource = check_serialization(resource)
+        return resource
+    except StopIteration:
+        return None
+
+
+class GenericResult:
     def __init__(self, results: Dict[Resource, HandlerContext]):
         self.results = results
 
-    def assert_all(self, status: ResourceState = ResourceState.deployed):
+    def assert_all(self, status):
         for r, ct in self.results.items():
             assert (
                 ct.status == status
-            ), f"Resource {r.id} has deploy status {ct.status}, expected {status}"
+            ), f"Resource {r.id} has status {ct.status}, expected {status}"
+
+    def assert_has_no_changes(self):
```

Comment:
type annotations are required on this repo, please use the PR template to see if all things are ok  https://github.com/inmanta/pytest-inmanta/blob/master/.github/pull_request_template.md